cmd/yesiscan/main.go

```diff
@@ -468,6 +573,80 @@ func GetConfig(configPath string) (*Config, error) {
 	return &configData, nil
 }
 
+// GetConfigPath returns the expected path to the main config.json file given
+// the input arg for that setting.
+func GetConfigPath(configPath string) (string, error) {
+	// If config path is set, we look in there for a config, otherwise we
+	// use the default xdg path.
+	if configPath != "" {
+		return configPath, nil
+	}
+
+	home, err := os.UserHomeDir()
+	if err != nil {
+		return "", errwrap.Wrapf(err, "error finding home directory")
+	}
+	if home == "" {
+		return "", fmt.Errorf("home directory is empty")
+	}
+
+	// TODO: get program from an input var perhaps?
+	p := filepath.Join(home, ".config/", program+"/", ConfigFileName)
+	return filepath.Clean(p), nil
+}
+
+// DownloadConfig pulls a config from a magic URI and returns the contents.
+func DownloadConfig(uri string) ([]byte, error) {
+	if uri == "" {
+		return nil, fmt.Errorf("empty URI")
+	}
+
+	u, err := url.Parse(uri)
+	if err != nil {
+		return nil, err
+	}
+
+	if u.Scheme == "https" {
+		client := &http.Client{
+			CheckRedirect: func() func(req *http.Request, via []*http.Request) error {
+				redirects := 0
+				return func(req *http.Request, via []*http.Request) error {
```

Comment:
This was the standard pattern I found, but if there is an improved version, please paste it here. Thanks

Reply:
Can you also confirm if I've understood it correctly? My purpose in posting this comment was to learn from you. Your response implies that I understood, but it would help me more if you explained where my understanding isn't quite right.